src/python/WMComponent/AgentStatusWatcher/AgentStatusPoller.py

```diff
@@ -168,6 +169,9 @@ def collectAgentInfo(self):
             if float(disk['percent'].strip('%')) >= diskUseThreshold and \
                             disk['mounted'] not in self.config.AnalyticsDataCollector.ignoreDisk:
                 agentInfo['disk_warning'].append(disk)
+                agentInfo['drain_mode'] = True
+                self.config.WorkQueueManager.queueParams['DrainMode'] = True
+                os.system("$manage execute-agent wmcoreD --restart --components=WorkQueueManager,AnalyticsDataCollector,AgentStatusWatcher")
```

Comment:
hmmm it looks like we already update the config each time but then modify it to the original config which we get at instantiation time `self.agentInfo` [AgentStatusPoller#L148](https://github.com/alexanderrichards/WMCore/blob/80448e32ccf09c57d017fa7550fbb16435649d07/src/python/WMComponent/AgentStatusWatcher/AgentStatusPoller.py#L148). I wonder if it's sufficient to simply remove the last line below. Why do we update this based on the config in the `self.agentInfo`?

```python
        logging.info("Getting agent info ...")
        agentInfo = self.wmagentDB.getComponentStatus(self.config)
        agentInfo.update(self.agentInfo)
```


Actually it looks like this `self.agentInfo` is only a couple of fields anyway [DataCollectAPI#L443](https://github.com/alexanderrichards/WMCore/blob/80448e32ccf09c57d017fa7550fbb16435649d07/src/python/WMComponent/AnalyticsDataCollector/DataCollectAPI.py#L443). Does this mean that we are already reloading the config?